compiledtrees/code_gen.py

```diff
@@ -18,15 +19,15 @@
 
 class CodeGenerator(object):
     def __init__(self):
-        self._lines = []
+        self._file = tempfile.NamedTemporaryFile(suffix='.cpp', delete=False, bufsize=0)
```

Comment:
The files are much smaller and sometimes the buffer doesn't write all of their's content to the disk. bufsize=0 disables the buffer. Although the better solution would be to trigger a write while the filename is returned.